File: mario.py
# build a half pyramid in (#) according to user input on height

from cs50 import get_int

# main function for half pyramid


def main():
    height = get_height()
    # rows of pyramid
    for i in range(height):
        # space in relation to height
        for space in range(height - i - 1):
            print(" ", end="")
        # columns of pyramid
        for j in range(i + 1):
            print("#", end="")

        print()

# request user input on height


def get_height():
    while True:
        n = get_int("Enter pyramid height(1-8 inclusive): ")
        # check user inputs an int 1-8 inclusive
        if n >= 1 and n <= 8:
            return n


main()

File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height, i, j, space;
    do
    //input height from user
    {
        height = get_int("Enter height of pyramid: ");
    }
    //ensure user inputs a value between 1 & 8
    while (height < 1 || height > 8);

    //rows of pyramid
    for (i = 0; i < height; i++)
    {
        //space in relation to height
        for (space = 0; space < height - i - 1; space++)
        {
            printf(" ");
        }
        //columns of pyramid
        for (j = 0; j <= i; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}